spec/models/miq_widget_spec.rb

```diff
@@ -175,24 +175,26 @@ def add_dashboard_for_user(db_name, userid, group)
         expect(@widget_report_vendor_and_guest_os.contents_for_user(@user1)).to eq(content)
       end
 
-      it "owned by miq_group and in user's timezone" do
+      it "returns widget contents in user's timezone when content from different timezone also available" do
         @user1.settings.store_path(:display, :timezone, "Eastern Time (US & Canada)")
-        content = FactoryBot.create(:miq_widget_content,
-                                     :miq_widget   => @widget_report_vendor_and_guest_os,
-                                     :miq_group_id => @group1.id,
-                                     :timezone     => "Eastern Time (US & Canada)"
-                                    )
-        expect(@widget_report_vendor_and_guest_os.contents_for_user(@user1)).to eq(content)
+        FactoryBot.create(:miq_widget_content,
+                          :miq_widget   => @widget_report_vendor_and_guest_os,
+                          :miq_group_id => @group1.id,
+                          :timezone     => "UTC")
+        content_user_timezone = FactoryBot.create(:miq_widget_content,
+                                                  :miq_widget   => @widget_report_vendor_and_guest_os,
+                                                  :miq_group_id => @group1.id,
+                                                  :timezone     => "Eastern Time (US & Canada)")
+        expect(@widget_report_vendor_and_guest_os.contents_for_user(@user1)).to eq(content_user_timezone)
       end
 
-      it "owned by miq_group and not in user's timezone" do
+      it "returns widget contents if only content available is not in user's timezone" do
```

Comment:
What happens if you have two or more contents for different timezones that aren't the user's timezone?  Will it always pick UTC first?  Will it be random?

For example, if user is Eastern time zone and we have content in UTC and pacific?  Which will we return UTC or pacific?

Reply:
according Rails doc - there is implied ordering